test/functional/lsp/callbacks/textDocument.lua

```diff
@@ -0,0 +1,38 @@
+local helpers = require('test.functional.helpers')(after_each)
```

Comment:
I guess this should be `test/functional/plugin/lsp/…`: other runtime files are tested in `plugin` unless these are runtime files used from core in a special fashion (like providers which have corresponding C code). Applies to all test files here.